src/Agent.Worker/Handlers/NodeHandler.cs

```diff
@@ -171,10 +171,16 @@ public async Task RunAsync()
         public string GetNodeLocation()
         {
             bool useNode10 = AgentKnobs.UseNode10.GetValue(ExecutionContext).AsBoolean();
+            bool useNode16 = AgentKnobs.UseNode16.GetValue(ExecutionContext).AsBoolean();
             bool taskHasNode10Data = Data is Node10HandlerData;
+            bool taskHasNode16Data = Data is Node16HandlerData;
 
             string nodeFolder = "node";
-            if (taskHasNode10Data || useNode10)
+            if (taskHasNode16Data || useNode16)
+            {
+                Trace.Info($"Task.json has node16 handler data: {taskHasNode16Data}");
+                nodeFolder = "node16";
+            } else if (taskHasNode10Data || useNode10)
```

Comment:
Please check the code formatting

```suggestion
            } 
            else if (taskHasNode10Data || useNode10)
```